Controllers/ImageController.cs

```diff
@@ -0,0 +1,336 @@
+﻿using System.IO.Compression;
+using System.Net;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+
+using AppImageGenerator.Models;
+
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.Formats.Webp;
+using SixLabors.ImageSharp.Formats.Bmp;
+using SixLabors.ImageSharp.Formats.Tiff;
+
+using Microsoft.AspNetCore.Mvc;
+
+using System.Net.Http.Headers;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+using System.Net.Http;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace AppImageGenerator.Controllers
+{
+    [ApiController]
+    [Route("api/")]
+    
+    public class ImageController : ControllerBase
+    {
+        private readonly IWebHostEnvironment _webHostEnvironment;
+
+        public ImageController (IWebHostEnvironment webHostEnvironment)
+        {
+            _webHostEnvironment = webHostEnvironment;
+        }
+
+        [HttpGet("getImagesZipById", Name = "getImagesZipById")]
+        public async Task<ActionResult> Get(string id)
+        {
+            try
+            {
+                // Create path from the id and return the file...
+                string zipFilePath = CreateFilePathFromId(new Guid(id));
+                if (string.IsNullOrEmpty(zipFilePath))
+                {
+                    var response = new NotFoundResult();
+                    return new NotFoundResult();
+                }
+
+                var archive = File(await System.IO.File.ReadAllBytesAsync(zipFilePath), "application/octet-stream", fileDownloadName: "AppImages.zip");
```

Comment:
Maybe pull some of these strings for things like `AppImages.zip` to the top as constants?

Reply:
Fixed